flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -216,7 +216,7 @@
     protected final Properties producerConfig;
 
     /** The name of the default topic this producer is writing data to. */
-    protected final String defaultTopicId;
+    protected String defaultTopicId;
```

Comment:
@Airblader , `defaultTopicId` would be initialised in the `invoke` method for the case that null value for `KafkaOption#TOPIC`.

Reply:
@twalthr , this is a valid change because `defaultTopicId` would be updated in the `invoke` method for the case that null value for `KafkaOption#TOPIC`.